### PR TITLE
Separate root and nonroot externs.

### PR DESCRIPTION
**Pull request by LucasSloan**

### CODE REVIEW

File: src/test/java/com/google/javascript/clutz/DepgraphTest.java [hunk 1]
```diff
@@ -25,6 +25,6 @@ public void testParseFile() throws Exception {
     assertThat(depgraph.getNonroots())
         .containsExactly("javascript/closure/base.js", "javascript/closure/string/string.js")
         .inOrder();
-    assertThat(depgraph.getExterns()).containsExactly("javascript/common/dom.js").inOrder();
+    assertThat(depgraph.getNonrootExterns()).containsExactly("javascript/common/dom.js").inOrder();
```
Comment (rkirov) [inline src/test/java/com/google/javascript/clutz/DepgraphTest.java]: Add assertion on getRootExterns too

Comment (rkirov) [reply src/test/java/com/google/javascript/clutz/DepgraphTest.java]: Add a test for partialInput flag branch too.

File: src/main/java/com/google/javascript/clutz/Depgraph.java [hunk 2]
```diff
@@ -38,7 +39,8 @@ Depgraph withNonrootsAsRoots() {
     Depgraph res = new Depgraph();
     res.roots.addAll(roots);
     res.roots.addAll(nonroots);
-    res.externs.addAll(externs);
+    res.rootExterns.addAll(rootExterns);
+    res.nonrootExterns.addAll(nonrootExterns);
```
Comment (mprobst) [inline src/main/java/com/google/javascript/clutz/Depgraph.java]: is this correct? it's oddly assymmetric.

File: src/test/java/com/google/javascript/clutz/DepgraphTest.java (hunk 1 above)
Comment (LucasSloan) [reply src/test/java/com/google/javascript/clutz/DepgraphTest.java]: Done, done.

File: src/main/java/com/google/javascript/clutz/Depgraph.java (hunk 2 above)
Comment (LucasSloan) [reply src/main/java/com/google/javascript/clutz/Depgraph.java]: @rkirov 

I've looked over the code, and currently nothing depends on the state of `rootExterns` or `nonrootExterns` after `withNonrootsAsRoots()` is called.  I agree with Martin that the less surprising state of affairs in the future is that all the externs are in `rootExterns`.

**Comment by LucasSloan** [comment]
I patched this into g3, and it did fix the error about files not being found.

File: src/main/java/com/google/javascript/clutz/Depgraph.java (hunk 2 above)
Comment (rkirov) [reply src/main/java/com/google/javascript/clutz/Depgraph.java]: yep, should be `res.rootExterns.addAll(nonrootExterns)'. This branch code is only executed when 'strictDeps' is off, and we usually have it on.

**Pull request merged**